subsys/caf/modules/shell.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/shell/shell.h>
+#include <app_event_manager.h>
+#include <caf/events/button_event.h>
+
+static int set_button_handler(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Missing required parameter(s)");
+		return -EINVAL;
+	}
+
+	struct button_event *event = new_button_event();
+
```

Comment:
For completeness, can you raise an error complaining about not enough heap if *event is NULL?

Reply:
the default allocator of events checks that already